### PR TITLE
[master < T1111] Fix local buffer serialization

### PR DESCRIPTION
**Pull request by as51340**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: tests/unit/storage_rocks.cpp [hunk 1]
```diff
@@ -9,111 +9,109 @@
...
-//   auto gid = edge->Gid();
-//   auto ser_result = memgraph::utils::SerializeEdge(vertex1.Gid(), vertex2.Gid(), edge->EdgeType(), edge->edge_.ptr);
-//   ASSERT_EQ(ser_result.first,
-//             memgraph::utils::SerializeIdType(vertex1.Gid()) + "|" + memgraph::utils::SerializeIdType(vertex2.Gid()) +
-//                 "|0|" + std::to_string(edge->EdgeType().AsInt()) + "|" + memgraph::utils::SerializeIdType(gid));
-//   ASSERT_EQ(ser_result.second,
-//             memgraph::utils::SerializeIdType(vertex2.Gid()) + "|" + memgraph::utils::SerializeIdType(vertex1.Gid()) +
-//                 "|1|" + std::to_string(edge->EdgeType().AsInt()) + "|" + memgraph::utils::SerializeIdType(gid));
-// }
+TEST_F(RocksDBStorageTest, SerializeVertexGIDLabels) {
```
Comment (Darych) [inline tests/unit/storage_rocks.cpp:63]: Should we also add tests for `SerializeVertexGIDLabelsOneProperty` and `SerializeVertexGIDLabelsManyProperties` ?

File: src/storage/v2/property_store.cpp [hunk 2]
```diff
@@ -1292,12 +1292,11 @@ void PropertyStore::SetBuffer(const std::string_view buffer) {
 
   uint64_t size = 0;
   uint8_t *data = nullptr;
+  size = buffer.size();
```
Comment (Darych) [inline src/storage/v2/property_store.cpp:1295]: Good catch! Does it mean that in loop on 1304 line we lost last byte value?

File: tests/unit/storage_rocks.cpp [hunk 3]
```diff
@@ -9,111 +9,109 @@
...
+}
+
+TEST_F(RocksDBStorageTest, SerializePropertiesExternalBuffer) {
+  memgraph::storage::PropertyStore props;
+  auto id = memgraph::storage::PropertyId::FromInt(0);
+  auto id_value = memgraph::storage::PropertyValue(1);
+  auto completion_percentage = memgraph::storage::PropertyId::FromInt(1);
+  auto completion_percentage_value = memgraph::storage::PropertyValue(14);
+  auto gender = memgraph::storage::PropertyId::FromInt(2);
+  auto gender_value = memgraph::storage::PropertyValue("man167863816386826");
```
Comment (Darych) [inline tests/unit/storage_rocks.cpp:103]: Maybe we can add a comment or empty lines to emphasize that because of this line we start using external buffer, what do you think? Because on the first glance it was hard to see the difference with previous test.

File: tests/unit/storage_rocks.cpp (hunk 1 above)
Comment (as51340) [reply tests/unit/storage_rocks.cpp:63]: I mean we can but I don't see what we would get more since properties are serialized as value and labels as key so I think that we wouldn't get nothing more than we have.

File: src/storage/v2/property_store.cpp (hunk 2 above)
Comment (as51340) [reply src/storage/v2/property_store.cpp:1295]: Yes and that was edge case :bug: :lady_beetle:

File: tests/unit/storage_rocks.cpp (hunk 3 above)
Comment (as51340) [reply tests/unit/storage_rocks.cpp:103]: Yes sure

File: tests/unit/storage_rocks.cpp (hunk 1 above)
Comment (Darych) [reply tests/unit/storage_rocks.cpp:63]: We will be sure that properties are listed through commas as we designed. If something will be changed in serialization function we will catch it.

Comment (as51340) [reply tests/unit/storage_rocks.cpp:63]: but properties are serialized just as with in-memory, no commas there?

Comment (as51340) [reply tests/unit/storage_rocks.cpp:63]: Can you explain what do you mean?

Comment (Darych) [reply tests/unit/storage_rocks.cpp:63]: Oops my bad. I messed with properties and labels. I meant to add multiple labels, not properties. You are right, properties are put as a serialized buffer.

**Review by Darych — APPROVED** [review]

**Comment by vpavicic** [comment]
@as51340 release note or is it an internal fix?

**Comment by as51340** [comment]
@vpavicic users actually have something out of it.
 Fixed bug when (de)serializing vertex and edge properties to RocksDB when the size of the serialization buffer was exactly 15B.
